src/utils/hooks/useUpdateAssignment.tsx

```diff
@@ -86,12 +95,20 @@ const createAssignmentPatch = (
   removedSections.forEach((code) => assignedSectionsCodes.delete(code))
   addedSections.forEach((code) => assignedSectionsCodes.add(code))
 
+  const trigger =
+    assignedSectionsCodes.size === 0
+      ? Trigger.OnReviewUnassign
+      : assignment.isSelfAssignable
+      ? Trigger.OnReviewSelfAssign
+      : Trigger.OnReviewAssign
```

Comment:
In our configs, are we actually distinguishing between "OnReviewSelfAssign" and "OnReviewAssign"?

Seems like there'd be quite a bit of duplication between them?

What is the use case when a Self-Assign trigger would cause different actions to an "Assign" trigger?

Reply:
Hmmm I had a look on that, and you are totally right! I think we can get rid of that event! I'll have to do some changes to the back end - and also to remove the old Trigger `Trigger.OnReviewSelfAssign` from core_templates but I'll do it in the Part 1 Back-end (so I don't need to add a new one)

Reply:
In that case, can we get rid of the "OnReviewSelfAssign" trigger altogether (i.e. remove it from possible Trigger enum values)?

Reply:
I tried, but it would mean it fails on snapshots that are still using it. Should it be safe to do?